src/Symfony/Component/ObjectMapper/Attribute/MapCollection.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ObjectMapper\Attribute;
+
+use Attribute;
+
+/**
+ * Maps an array of associative data to an array of typed objects.
+ *
+ * Example:
+ *     #[MapCollection(of: ProductDTO::class)]
+ *     public array $products;
+ *
+ * @experimental
+ *
+ * @author Devoton <oton.traore@email.com>
+ */
+#[Attribute(Attribute::TARGET_PROPERTY)]
+class MapCollection
```

Comment:
I'm :-1: for this as I don't really see the point, could you give me a real life example where you'd want something like this? 

If we start supporting arrays in the ObjectMapper it can become quite complex, we should instead use a CollectionMapper but I'm quite not sure of the functional benefits behind it.